packages/mdc-tab-indicator/icon-foundation.js

```diff
@@ -0,0 +1,37 @@
+/**
+  * @license
+  * Copyright 2018 Google Inc. All Rights Reserved.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License")
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *      http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+*/
+
+import MDCTabIndicatorFoundation from './foundation';
+
+/**
+ * @final
+ */
+class MDCTabIndicatorIconFoundation extends MDCTabIndicatorFoundation {
```

Comment:
I thought maybe I would see some logic around delaying the fade in and fade out. But I don't see anything in this file like that.

I wonder if it should be

```
activate() {
  super.activate();
  this.adapter_.addClass(MDCTabIndicatorFoundation.cssClasses.ANIMATING_ICON_ACTIVATE);
}
```

Reply:
The delayed fade-in is in the CSS `transition-end` property.

The reason I avoided making the `activate` and `deactivate` concrete methods and instead made them abstract was to provide more granular control over when the handlers should be added or when to make a the indicator active.

That being said, I can make them concrete methods.